Pod/Classes/XExtensionItem.m

```diff
@@ -24,18 +24,25 @@ @implementation XExtensionItemSource
 
 #pragma mark - Initialization
 
-- (instancetype)initWithPlaceholderItem:(id)placeholderItem typeIdentifier:(NSString *)typeIdentifier itemBlock:(XExtensionItemProvidingBlock)activityItemBlock {
+- (instancetype)initWithPlaceholderItem:(id)placeholderItem
+                         typeIdentifier:(NSString *)typeIdentifier
+                              itemBlock:(XExtensionItemProvidingBlock)activityItemBlock {
     NSParameterAssert(placeholderItem);
+    NSParameterAssert(typeIdentifier);
     
     self = [super init];
     if (self) {
         _placeholderItem = placeholderItem;
         _activityItemBlock = [activityItemBlock copy];
         
-        if (typeIdentifier)
-            _typeIdentifier = [typeIdentifier copy];
-        else
-            _typeIdentifier = [typeIdentifierForActivityItem(placeholderItem) copy];
+        _typeIdentifier = ^{
+            if (typeIdentifier) {
+                return [typeIdentifier copy];
+            }
+            else {
+                return [typeIdentifierForActivityItem(placeholderItem) copy];
+            }
+        }();
```

Comment:
I've never seen this style before and I'm very curious why a block seems preferable. To my eyes it seems considerably less clean (and more overhead) than just using an if statement.